### PR TITLE
Ci: refactor pip install logic for test and lint fix #26

### LINKED ISSUE #26: Ci: GitHub Actions workflow for test and lint can be cleaner
we do not need to generate requirements and  pip install for this.

### PR DESCRIPTION
**Pull request by yihong0618**
fix: #26 
refactor pip install logic for test and lint
for `pyproject.toml`  repo there is an easy way to install
this `PR` make the GitHub Actions cleaner.
my passed job can check https://github.com/yihong0618/pgtracer/actions/runs/3367072889/jobs/5584216925
the reason this Actions failed is that I can not upload coverage reports to `Codecov`

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -27,13 +27,8 @@ jobs:
...
           # Must be installed via the system
           sudo apt install python3-bpfcc python3-pip
-          pip install setuptools toml
-          # Install requirements from pyproject.toml
-          python -c 'import toml; open("requirements.txt.tmp", "w").write("\n".join(toml.load(open("pyproject.toml"))["project"]["dependencies"]) + "\n")'
-          # Install lint requirements
-          python -c 'import toml; open("requirements.txt.tmp", "a").write("\n".join(toml.load(open("pyproject.toml"))["project"]["optional-dependencies"]["lint"]) + "\n")'
-          pip install -r requirements.txt.tmp
-          python -c "import bcc; print(bcc.__file__)"
+          pip install -U pip toml
```
Comment (yihong0618) [inline .github/workflows/lint.yml:30]: install the latest pip contains setuptools

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -44,18 +44,14 @@ jobs:
...
-          # Install lint requirements
-          python -c 'import toml; open("requirements.txt.tmp", "a").write("\n".join(toml.load(open("pyproject.toml"))["project"]["optional-dependencies"]["lint"]) + "\n")'
-          sudo pip install -r requirements.txt.tmp
+          sudo pip install '.[test]'
           python -c "import bcc; print(bcc.__file__)"
 
       - id: tests
         run: |
-          sudo PYTHONPATH=$(pwd)/src pytest --postgresql-exec /usr/lib/postgresql/${{matrix.postgresql_version}}/bin/pg_ctl --cov --cov-report=xml
+          sudo pytest --postgresql-exec /usr/lib/postgresql/${{matrix.postgresql_version}}/bin/pg_ctl --cov --cov-report=xml
```
Comment (yihong0618) [inline .github/workflows/tests.yml:54]: no need set PYTHONPATH here since we install the pgtracer package

File: .github/workflows/lint.yml (hunk 1 above)
Comment (rdunklau) [reply .github/workflows/lint.yml:30]: Great I didn't know about this !

**Comment by yihong0618** [comment]
upload `Codecov` seems flanky.

**Comment by rdunklau** [comment]
Hello, thank you for this PR !

Before I merge it, I have to bring to your attention that we just changed the license to GPL V3. If that is not a problem for you I will merge this.

**Comment by yihong0618** [comment]
> merge it, I

No problem! thanks

**Pull request merged**